threat_intel/util/http.py

```diff
@@ -275,11 +275,18 @@ def _wait_for_response(self, requests, to_json):
             A list of dicts if to_json, a list of grequest.response otherwise
         """
         all_responses = []
+        auth_error = False
 
         for retry in range(self._max_retry):
             try:
                 responses = grequests.map(requests)
                 valid_responses = [response for response in responses if response]
+                failed_auth_responses = [response for response in responses if response.status_code == 403]
+
+                if len(failed_auth_responses) > 0:
+                    auth_error = True
```

Comment:
Any chance to catch a more specific error in line 294?
Catching all errors looks ugly in case...
Although I guess here it is very much appreciated ;)